library/models/library.py

```diff
@@ -256,19 +267,49 @@ def on_change_card_issue(self, card_id):
             @return : Dictionary having identifier of the record as key
                       and the user info as value
         '''
-        if not card_id:
-            return {'value': {}}
-        card_data = self.env['library.card'].browse(card_id)
-        val = {'user': str(card_data.user.title())}
-        if card_data.user.title() == 'Student':
-            val.update({'student_id': card_data.student_id.id,
-                        'standard_id': card_data.standard_id.id,
-                        'roll_no': card_data.roll_no,
-                        'gt_name': card_data.gt_name})
-        else:
-            val.update({'teacher_id': card_data.teacher_id.id,
-                        'gt_name': card_data.gt_name})
-        return {'value': val}
+        if self.card_id:
+            self.user = str(self.card_id.user.title()) or ''
+            if self.card_id.user.title() == 'Student':
+                self.student_id = self.card_id.student_id.id or False
+                self.standard_id = self.card_id.standard_id.id or False
+                self.roll_no = int(self.card_id.roll_no) or False
+                self.gt_name = self.card_id.gt_name or ''
+
+            else:
+                self.teacher_id = self.card_id.teacher_id.id
+                self.gt_name = self.card_id.gt_name
+
+    @api.model
+    def create(self, vals):
+        if vals.get('card_id') and vals.get('user') == 'Student':
```

Comment:
Here we are checking that card holder type is student or teacher. So we check the selection field with static condition.